src/Agent.Worker/JobRunner.cs

```diff
@@ -308,7 +312,38 @@ public sealed class JobRunner : AgentService, IJobRunner
 
                 Trace.Info($"Job result after all post-job steps finish: {jobContext.Result}");
 
-                // Complete the job.
+                // If missing, will default to false.
+                bool uploadDiagnosticLogs = false;
+                if (message.Environment.Variables.ContainsKey(Constants.Variables.Agent.Diagnostic))
+                {
+                    uploadDiagnosticLogs = StringUtil.ConvertToBoolean(message.Environment.Variables[Constants.Variables.Agent.Diagnostic]);
+                }
+
+                if (uploadDiagnosticLogs)
+                {
+                    Trace.Info("Support log upload starting.");
+
+                    IDiagnosticLogManager diagnosticLogManager = HostContext.GetService<IDiagnosticLogManager>();
+
+                    try
+                    {
+                        diagnosticLogManager.UploadDiagnosticLogs(
```

Comment:
do we want to do this here, or make the diag log upload as a step?

Reply:
Makes sense, we could add it conditionally as a post job step. Is that what you mean?

Reply:
yes, basically be a node in the timeline record tree, 
you can just create a node that just for the log upload.

Reply:
Let's discuss this tomorrow. I don't think Bryan and Chris want there to be an extra node in the tree. The UI is supposed to show the links to download the zip files in the Diagnostic logs section on the build summary page. I can forward you the latest UI design.